src/odemis/driver/test/pwrmccdaq_test.py

```diff
@@ -0,0 +1,175 @@
+        time.sleep(pwrmccdaq.INTERLOCK_POLL_INTERVAL)  # wait just long enough until the next thread's interval
+        self.assertTrue(self.mcc_device.interlockTriggered.value)
+        # check if the bit status is now the new bit status
+        new_bit_status = self.mcc_device.device.DIn(port)
+        self.assertNotEqual(old_bit_status, new_bit_status)
+
+    def test_incorrect_di_list(self):
+        """
+        Test for the instantiation of a basic MCC device with incorrect DI list values or structure.
+        Also does a small check for the poll thread to not have started when there are no status
+        DI's to check.
+        """
+        MOD_CONFIG_MCC = copy.deepcopy(CONFIG_MCC)
+        MOD_CONFIG_MCC["di_channels"] = {2: ["interlockTriggered", False], 99: ["testPropery", True]}
+        with self.assertRaises(ValueError):
+            pwrmccdaq.MCCDevice(**MOD_CONFIG_MCC)
+
+        MOD_CONFIG_MCC = copy.deepcopy(CONFIG_MCC)
+        MOD_CONFIG_MCC["di_channels"] = {11: [True, "testPropery"]}
+        with self.assertRaises(TypeError):
+            pwrmccdaq.MCCDevice(**MOD_CONFIG_MCC)
+
+        # check list with no channel values given
+        MOD_CONFIG_MCC = copy.deepcopy(CONFIG_MCC)
+        MOD_CONFIG_MCC["di_channels"] = None
+        self.mcc_device = pwrmccdaq.MCCDevice(**MOD_CONFIG_MCC)
+
+        # check if there are no extra Boolean VA's
+        for va in model.getVAs(self.mcc_device):
+            self.assertFalse(isinstance(va, model.BooleanVA))
+
+        # be sure that there is no poll tread started as there are no Boolean VA's to check
+        self.assertFalse(self.mcc_device._status_thread)
+
+    def test_thread_termination(self):
+        """
+        Check if the polling thread for the status changes of the DI's is working properly.
+        """
+        self.mcc_device = pwrmccdaq.MCCDevice(**CONFIG_MCC)
+
+        # check if polling of the DI port bits has started
+        self.assertTrue(self.mcc_device._status_thread.is_alive())
+        self.assertFalse(self.mcc_device._status_thread.terminated)
+
+        self.mcc_device.terminate()
+        time.sleep(0.5)  # give the thread some time to suspend
+
+        # check if polling of the DI port bits has suspended
+        self.assertTrue(self.mcc_device._status_thread.terminated)
+        self.assertFalse(self.mcc_device._status_thread.is_alive())
+
+    def tearDown(self):
+        if self.mcc_device._status_thread:
+            self.mcc_device.terminate()
+
+
+class TestMCCDeviceLight(unittest.TestCase):
+    # test for power setting normal and under- and overlimit, really fast power requests
+    #   very tiny or very big power requests?
+    # wrong channel/spectra/pwr_curve number, e.g. spectra not logical
+    # check subscribing and unsubscribing from power VA
+    # check right analog output voltage and if DIO is set accordingly
+
+    def setUp(self):
+        pass
+
+    def tearDown(self):
+        pass
+
+    def test_power(self):
```

Comment:
I think you can mostly copy paste the tests from other drivers, like the pwrcomedi_test.py ones.